### ISSUE TITLE
picking up deleted submissions

### ISSUE THREAD

**Issue by L-F-Escobar** [dmarx/psaw#31]
redditBot2 = praw.Reddit('bot2')
     apiBot2 = psaw.PushshiftAPI(r=redditBot2)
     submissions = apiBot2.search_submissions(limit=2,  subreddit="bottesting", q='insert_term')

     for submission in submissions:
          # printing submission meta data here'

     for comment in submission.comments:
          # printing comment meta data here

I get scopping up deleted submissions. I found a potential solve but it didnt work

       author=![deleted],
- https://redd.it/9cfmo1

I do not see documentation how to deal with this issue. I am always passing a current instance of reddit into redditBot2.

**Comment by dmarx** [comment]
> I am always passing a current instance of reddit into redditBot2

Well there's your problem ;)

I think it'll help if I explain how the different components of this system work together. 

1. **pushshift**. Pushshift is an external database that archives public reddit activity. It stays thorough by requesting incremental IDs rather than just streaming the /r/all feed. This means pushshift has some minimal detail about content it can't actually access, e.g. content in private subs. Older content (prior to 2015ish?) Was backfilled, but otherwise you should expect that most attributes reflect the state of the content within a few seconds of when it was submitted, with some attributes like score updated a month later. This is why you're able to read the text of deleted comments pulled from pushshift.

2. **psaw** is just a convenient way to interact with the database. The default behavior is to request content directly from pushshift without interacting with reddit.com at all.

3. **praw** is a convenient way to interact with reddit.com's api. When you give psaw a praw.Reddit instance, psaw will query pushshift just for object IDs, then uses praw to request the associated content. Because praw is the component requesting the content, it will download the most current version of that object. If a public comment was deleted, it's likely that pushshift has a snapshot of what it says, but if you give psaw a praw instance you're not going to see it.

TLDR: just don't give psaw a praw instance.

**Comment by L-F-Escobar** [comment]
That seems to have solved my issue but the data returned is a much different format now so I cant tell just by looking at it because it seems that many json key:value pairs are being returned now. 

I stopped sending in a psar instance into a psaw. I still dont entirely get why that was causing my issue. 


Edit: So upon further review I am still getting back a deleted submission.

My modified code look like this 

     apiBot2 = psaw.PushshiftAPI()

I am no longer passing a praw object into my psaw object. How would I comment on things though if the bot doesnt have any authentication? Also, my original issue is not solved. The link to the deleted post I am picking up is 

-  full_link='https://www.reddit.com/r/bottesting/comments/9d1srx/test/

**Comment by dmarx** [comment]
Because when you hit pushshift, you're requesting an archival record in whatever format the author of the project chose to store their data, whereas when you use praw you're getting whatever the current state of that item is. 

When you were seeing "[deleted]" comments, it's because the author deleted their comment and that's what you would see if you visited that comment on reddit.com. When you use psaw without praw, you get to see what the comment said before it was deleted because you're getting it from a pre-scraped dataset.

> I am no longer passing a praw object into my psaw object. How would I comment on things though if the bot doesnt have any authentication? 

Psaw doesn't have commenting functionality, that's just praw. You could use psaw to monitor reddit and then pass the content id to the authenticated praw to comment.

Psaw (without praw) doesn't interact with reddit.com at all, it's just a convenience wrapper for accessing the REST API offered by pushshift.io. You're getting reddit content through a middle man. Someone else scraped reddit, put comments and posts into a database, and is letting you access this collection. They needed to be authenticated for their scrape of reddit, but you don't need to be authenticated for your scrape of their dataset.

Let me know if that clarifies what's going on. Closing since root cause was addressed, but feel free to comment here if you have further questions about this. It would help if you described your bot more: I have no idea what you're trying to do here.
